enhancements/installer/credentials-management-outside-openshift-cluster.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Credentials Management outside OpenShift Cluster
+authors:
+  - "@akhil-rane"
+reviewers:
+  - "@abhinavdahiya"
+  - "@dgoodwin"
+  - "@joelddiaz"
+approvers:
+  - "@derekwaynecarr"
+  - "@sdodson"
+creation-date: 2021-03-25
+last-updated: 2021-03-25
+status: provisional
+---
+
+# Credentials Management outside OpenShift Cluster
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The intent of this enhancement is to take the process of credentials management outside the OpenShift cluster for new 
+platforms. We propose to make *manual* mode as default for clusters on new platforms. 
+
+## Motivation
+
+The main motivation behind this enhancement is to satisfy the customer requirement to follow the best security practices 
+by not storing admin credentials inside the cluster
+
+### Goals
+
+As part of this enhancement we plan to do the following:
+* Set *manual* as default/preferred credentials mode for OpenShift
+* Not store admin level cloud credentials inside a cluster
+* Provide a tooling design for credentials management outside the cluster
+
+### Non-Goals
+
+* We do not plan to go into design details on how this tool will be ported to cloud providers other than AWS.
+
+## Proposal
+
+Currently, *mint* is a default credentials mode for OpenShift. In this mode we run OpenShift installer with an admin 
+level cloud credential. The admin credential is stored in kube-system namespace and then used by the cloud credential 
+operator to process the CredentialRequests in the cluster and create new users with fine-grained permissions.
+The customers have reported that the need to store admin credentials inside the cluster is a major disadvantage. Based 
+on the feedback, we propose to run credentials related setup outside a cluster and then start installation process in a 
+*manual* mode without providing admin credentials to the installer.
+
+### User Stories
+
+#### Story 1
+As a OpenShift cluster administrator I should be able to extract the CredentialsRequest manifests from the release image
```

Comment:
Is the `ccoctl` tool going to be tied to a release image? Or will the user need to know the release image?

Reply:
We expect that it will be tied to a release image and extracted with oc release commands. I don't think it will be tightly coupled in that it would often work with future releases fine.

Reply:
That doesn't quite answer my (poorly phrased) question. When using `ccoctl`, is the expectation that the user would need to specify the release image from which to get the CredentialsRequests?

Reply:
Ah yes, for install prep yes you will need to refer to the release image with the cred requests you will need. I am hoping we can run day 2 and work against the CredRequests in a live cluster/current kube context to help users reconcile.